### PR TITLE
{data}[GCCcore/12.3.0] HDF v4.2.16-2

### PR DESCRIPTION
**Pull request by stefan-wolfsheimer**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by stefan-wolfsheimer** [comment]
I could not test if the software builds on ARM. 
The patch introduced in 4.2.15 does not work anymore 4.2.16-2.

**Comment by branfosj** [comment]
> I could not test if the software builds on ARM. The patch introduced in 4.2.15 does not work anymore 4.2.16-2.

I searched through the commits and looked at the HDF4 code. It appears that they've made a big change to how different architectures are handled and I believe this means that ARM will not need a patch.

**Comment by branfosj** [comment]
@boegelbot please test @ generoso

File: easybuild/easyconfigs/h/HDF/HDF-4.2.16-2-GCCcore-12.3.0.eb [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+configopts = [
+    '--with-szlib=$EBROOTSZIP CFLAGS="$CFLAGS -I$EBROOTLIBTIRPC/include/tirpc"',
+    '--includedir=%(installdir)s/include/%(namelower)s FFLAGS="$FFLAGS -fallow-argument-mismatch"',
+    '--with-szlib=$EBROOTSZIP CFLAGS="$CFLAGS -I$EBROOTLIBTIRPC/include/tirpc"',
+    '--includedir=%(installdir)s/include/%(namelower)s',
+    '--enable-shared',
+    '--disable-fortran',
+    '--disable-netcdf',
+]
+
```
Comment (branfosj) [inline easybuild/easyconfigs/h/HDF/HDF-4.2.16-2-GCCcore-12.3.0.eb:44]:
```suggestion
local_common_configopts = '--with-szlib=$EBROOTSZIP CFLAGS="$CFLAGS -I$EBROOTLIBTIRPC/include/tirpc" '
local_common_configopts += '--includedir=%(installdir)s/include/%(namelower)s '
configopts = [
    local_common_configopts,
    # Cannot build shared libraries and Fortran...
    # https://trac.osgeo.org/gdal/wiki/HDF#IncompatibilitywithNetCDFLibraries
    # netcdf must be disabled to allow HDF to be used by GDAL
    local_common_configopts + "--enable-shared --disable-fortran --disable-netcdf",
]
```

With `configopts` as a list we end up building once per item in the list.

**Review by branfosj — CHANGES_REQUESTED** [review]

**Comment by SebastianAchilles** [comment]
Test report by @sebastianachilles
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
bwd-rockylinux-92 - Linux Rocky Linux 9.2 (Blue Onyx), x86_64, Intel(R) Core(TM) i7-6900K CPU @ 3.20GHz (broadwell), Python 3.9.16
See https://gist.github.com/SebastianAchilles/d85deaa8bfbfa008806247497fccb961 for a full test report.

**Comment by SebastianAchilles** [comment]
@boegelbot please test @ generoso

**Comment by SebastianAchilles** [comment]
@boegelbot please test @ jsc-zen2

**Review by SebastianAchilles — APPROVED** [review]
lgtm

**Comment by SebastianAchilles** [comment]
Going in, thanks @stefan-wolfsheimer!

**Pull request merged**